### PR TITLE
Improved AWS cred-req flow, create cred-reqs only for tests that require them

### LINKED ISSUE #2542: CredentialsRequest check times out and leaves bucket after running
The current 40 seconds dedicated to checking the credentials do not seem enough, thus leading to test failures because of timeouts.
Also, sometimes, buckets are left over after the check ran - we need to find a better way to check the credentials (preferably one that is also not dependent on bucket creation)

### PR DESCRIPTION
**Pull request by MeridianExplorer**
Closes #2542

### CODE REVIEW

File: ocs_ci/ocs/resources/mcg.py [hunk 1]
```diff
@@ -117,7 +117,10 @@ def __init__(self):
 
         self.s3_client = self.s3_resource.meta.client
 
-        if config.ENV_DATA['platform'].lower() == 'aws':
+        if (
+            config.ENV_DATA['platform'].lower() == 'aws'
+            and kwargs.get('create_aws_creds')
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py:122]: what if `create_aws_creds` doesn't exist in kwargs? Please add a default:

```python
kwargs.get('create_aws_creds', 'default value')
```

File: tests/conftest.py [hunk 2]
```diff
@@ -1546,21 +1546,32 @@ def mcg_obj_session(request):
...
 
-    mcg_obj = MCG()
+    mcg_obj = MCG(*args, **kwargs)
 
     def finalizer():
         if config.ENV_DATA['platform'].lower() == 'aws':
             mcg_obj.cred_req_obj.delete()
 
-    request.addfinalizer(finalizer)
+    if kwargs.get("create_aws_creds"):
```
Comment (ebenahar) [inline tests/conftest.py:1573]: same, add a default value

File: ocs_ci/ocs/resources/mcg.py (hunk 1 above)
Comment (MeridianExplorer) [reply ocs_ci/ocs/resources/mcg.py:122]: `.get()`returns `None` by default, which will be evaluated as `False` in this scenario - so this is working as intended

File: tests/conftest.py (hunk 2 above)
Comment (MeridianExplorer) [reply tests/conftest.py:1573]: Ditto

File: ocs_ci/ocs/resources/mcg.py (hunk 1 above)
Comment (ebenahar) [reply ocs_ci/ocs/resources/mcg.py:122]: ack, thanks

**Pull request merged**